### PR TITLE
Replace k10-generate-gold-backup-policy ClusterPolicy

### PR DESCRIPTION
**Pull request by mattbator**
## Related Issue(s)

N/A

## Description

Replaces `k10-generate-gold-backup-policy` with a version that checks for duplicate policies and can more easily be adapted to environmental needs. Filenames, policy/rule names, category, and keywords have been updated to reflect Kasten K10 rebranding to Veeam Kasten.

## Checklist

- [x] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [x] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.
- [x] I have added the artifacthub-pkg.yml file and have verified it is complete and correct.

### CODE REVIEW

File: kasten/kasten-generate-example-backup-policy/artifacthub-pkg.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+version: 1.0.1
+displayName: Generate Kasten Backup Policy Based on Resource Label
+createdAt: "2023-05-07T00:00:00.000Z"
+description: >-
+  Generates a Kasten policy for a namespace that includes any Deployment or StatefulSet with a "dataprotection=kasten-example" label, if the policy does not already exist.
+
+  Use with "kasten-data-protection-by-label" policy to require "dataprotection" labeling on workloads.
+install: |-
+  ```shell
+  kubectl apply -f https://raw.githubusercontent.com/kyverno/policies/main/kasten/kasten-generate-example-backup-policy/create-kasten-policies-clusterrole.yaml
```
Comment (chipzoller) [inline kasten/kasten-generate-example-backup-policy/artifacthub-pkg.yml:11]:
```suggestion
```

File: kasten/kasten-generate-example-backup-policy/artifacthub-pkg.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  ```
+keywords:
+  - kyverno
+  - Veeam Kasten
+readme: |
+  Generates a Kasten policy for a namespace that includes any Deployment or StatefulSet with a "dataprotection=kasten-example" label, if the policy does not already exist.
+
+  Use with "kasten-data-protection-by-label" policy to require "dataprotection" labeling on workloads.
+  
+  Refer to the documentation for more details on Kyverno annotations: https://artifacthub.io/docs/topics/annotations/kyverno/
```
Comment (chipzoller) [inline kasten/kasten-generate-example-backup-policy/artifacthub-pkg.yml:21]:
```suggestion
```

File: kasten/kasten-generate-example-backup-policy/kasten-generate-example-backup-policy.yaml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    policies.kyverno.io/title: Generate Kasten Backup Policy Based on Resource Label
+    policies.kyverno.io/category: Veeam Kasten
+    kyverno.io/kyverno-version: 1.12.1
+    policies.kyverno.io/minversion: 1.12.0
+    kyverno.io/kubernetes-version: "1.24-1.30"
+    policies.kyverno.io/subject: Policy
+    policies.kyverno.io/description: >-
+      Generates a Kasten policy for a namespace that includes any Deployment or StatefulSet with a "dataprotection=kasten-example" label, if the policy does not already exist.
+
+      Use with "kasten-data-protection-by-label" policy to require "dataprotection" labeling on workloads.
```
Comment (chipzoller) [inline kasten/kasten-generate-example-backup-policy/kasten-generate-example-backup-policy.yaml:23]: When using a folded line (block) scalar like this, you can't have an empty line or else it breaks rendering.
```suggestion
      Generates a Kasten policy for a namespace that includes any Deployment or StatefulSet with a "dataprotection=kasten-example" label, if the policy does not already exist. Use with "kasten-data-protection-by-label" policy to require "dataprotection" labeling on workloads.
```

**Review by chipzoller — CHANGES_REQUESTED** [review]
One minor thing. I noticed when you had done this to previous samples, it broke our ability to render them to Markdown :)

**Comment by chipzoller** [comment]
Ping @mattbator

**Comment by mattbator** [comment]
Thanks for the support on changes required to play nicely with the ArtifactHub - should be in good shape now.

**Comment by mattbator** [comment]
Good question! The original contribution was an early policy created by my predecessor prior to any test requirements. I was able to fumble my way through chainsaw updates to the other policies with minor changes, but believe I may have been a bit out of my depth on this one. Happy to take another pass at it if they're now required.

**Comment by chipzoller** [comment]
Thanks for the info. I think this is fine.

**Review by chipzoller — APPROVED** [review]

**Comment by mattbator** [comment]
Thank you!

**Pull request merged**